docs/src/main/asciidoc/secretmanager.adoc

```diff
@@ -0,0 +1,83 @@
+A detailed summary of its features can be found in the https://cloud.google.com/blog/products/identity-security/introducing-google-clouds-secret-manager[Secret Manager documentation].
+
+Spring Cloud GCP provides:
+
+* A Spring Boot starter which automatically loads the secrets of your GCP project into your application context as a https://cloud.spring.io/spring-cloud-commons/multi/multi__spring_cloud_context_application_context_services.html#_the_bootstrap_application_context[Bootstrap Property Source].
+* A `SecretManagerTemplate` which allows you to read, write, and update secrets in Secret Manager.
+
+=== Dependency Setup
+
+To begin using this library, add the `spring-cloud-gcp-starter-secretmanager` artifact to your project.
+
+Maven coordinates, using <<getting-started.adoc#_bill_of_materials, Spring Cloud GCP BOM>>:
+
+[source,xml]
+----
+<dependency>
+  <groupId>org.springframework.cloud</groupId>
+  <artifactId>spring-cloud-gcp-starter-secretmanager</artifactId>
+</dependency>
+----
+
+Gradle coordinates:
+
+[source]
+----
+dependencies {
+  compile group: 'org.springframework.cloud', name: 'spring-cloud-gcp-starter-secretmanager'
+}
+----
+
+=== Secret Manager Property Source
+
+The Spring Cloud GCP integration for Google Cloud Secret Manager enables you to use Secret Manager as a bootstrap property source.
+
+This feature allows you to automatically load your GCP project's secrets as properties into the application context during the https://cloud.spring.io/spring-cloud-commons/reference/html/#the-bootstrap-application-context[Bootstrap Phase], which refers to the initial phase when a Spring application is being loaded.
+
+NOTE: To begin using this feature, you must set the setting `spring.cloud.gcp.secretmanager.bootstrap.enabled` to **true**, this feature disabled by default.
+
+Spring Cloud GCP will load the **latest** version of each secret into the application context.
+
+All secrets will be loaded into the application environment using their `secretId` as the property name.
+For example, if your secret's id is `my-secret` then it will be accessible as a property using the name `my-secret`.
+If you would like to append a prefix string to all property names imported from Secret Manager, you may use the `spring.cloud.gcp.secretmanager.secret-name-prefix` setting described below.
+
+Spring Cloud GCP Secret Manager offers several configuration properties to customize the behavior.
+
+|===
+| Name | Description | Required | Default value
+| `spring.cloud.gcp.secretmanager.bootstrap.enabled` | Enables loading secrets from Secret Manager as a bootstrap property source. Set this to **true** to enable the feature. | No | `false`
+| `spring.cloud.gcp.secretmanager.credentials.location` | OAuth2 credentials for authenticating with the Google Cloud Secret Manager API. | No | By default, infers credentials from https://cloud.google.com/docs/authentication/production[Application Default Credentials].
```

Comment:
I am not sure I can english properly today, so check that this makes sense grammatically.

```suggestion
| `spring.cloud.gcp.secretmanager.credentials.location` | OAuth2 credentials for authenticating to the Google Cloud Secret Manager API. | No | By default, infers credentials from https://cloud.google.com/docs/authentication/production[Application Default Credentials].
```

Reply:
Done.